File: Praktikum_Minggu_4/AddExpression.cpp
#include "AddExpression.hpp"
#include <iostream>

using namespace std;

AddExpression::AddExpression(Expression *x, Expression *y) : BinaryExpression(x, y) {
	this->x = x;
	this->y = y;
}

int AddExpression::solve() {
	return x->solve() + y->solve();
}

File: Tutorial_Minggu_7/Expression/AddExpression.java
class AddExpression implements Expression {
	protected Expression x;
	protected Expression y;

	public AddExpression(Expression x, Expression y) {
		this.x = x;
		this.y = y;
	}

	public int solve() {
		return this.x.solve() + this.y.solve();
	}
}